src/crawlee/http_clients/base_http_client.py

```diff
@@ -2,71 +2,97 @@
 
 from abc import ABC, abstractmethod
 from dataclasses import dataclass
-from typing import TYPE_CHECKING, Iterable, Protocol
+from typing import TYPE_CHECKING, Protocol
 
 if TYPE_CHECKING:
-    from httpx import Headers  # Type from `httpx` is used here because it is lightweight and convenient
-
     from crawlee.models import Request
     from crawlee.proxy_configuration import ProxyInfo
     from crawlee.sessions import Session
     from crawlee.statistics import Statistics
+    from crawlee.types import HttpMethod
 
 
 class HttpResponse(Protocol):
-    """Response to an HTTP call."""
+    """This protocol defines the interface that any HTTP response object must implement."""
 
     def read(self) -> bytes:
         """Read the content of the response body."""
 
     @property
     def status_code(self) -> int:
-        """HTTP status code of the response (https://developer.mozilla.org/en-US/docs/Web/HTTP/Status)."""
+        """The HTTP status code received from the server."""
 
     @property
-    def headers(self) -> Headers:
-        """Response headers."""
+    def headers(self) -> dict[str, str]:
```

Comment:
That's probably a good call. However, `httpx.Headers` has some subtle differences from `dict`, such as being case-insensitive. This may lead to some annoying bugs when switching HTTP clients. Could we make a `Mapping` subclass that lowercases the header names? Also this should probably be immutable...

Reply:
I added a new type `HttpHeaders`.